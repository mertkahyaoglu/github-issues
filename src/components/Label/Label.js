import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Label = ({ label }) => (
  <StyledLabel target="_blank" href={label.url} color={label.color}>
    {label.name}
  </StyledLabel>
);

Label.propTypes = {
  label: PropTypes.shape({
    color: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default Label;

const StyledLabel = styled.a`
  height: 20px;
  padding: 4px;
  font-size: 12px;
  font-weight: 600;
  line-height: 15px;
  border-radius: 2px;
  box-shadow: inset 0 -1px 0 rgba(27, 31, 35, 0.12);
  background-color: #${props => props.color};
  color: #000;
  margin-right: 4px;
`;
