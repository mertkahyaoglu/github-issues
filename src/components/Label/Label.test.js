import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Label from './Label';
import TestProviders from 'utils/TestProviders';

const label = {
  id: 418438082,
  url: 'https://api.github.com/repos/jekyll/jekyll-admin/labels/stale',
  name: 'stale',
  color: 'bfd4f2',
  default: false,
};

describe('component/Status', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <TestProviders>
        <Label label={label} />
      </TestProviders>,
      div
    );
  });

  it('renders markup correctly', () => {
    const tree = renderer
      .create(
        <TestProviders>
          <Label label={label} />
        </TestProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
