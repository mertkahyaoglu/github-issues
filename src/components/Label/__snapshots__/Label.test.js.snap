// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Status renders markup correctly 1`] = `
<a
  className="sc-bdVaJa bNnuZ"
  color="bfd4f2"
  href="https://api.github.com/repos/jekyll/jekyll-admin/labels/stale"
  target="_blank"
>
  stale
</a>
`;
